test/rpc_test.py

```diff
@@ -1044,6 +1044,26 @@ def test_set_rpc_timeout(self):
         self.assertEqual(timeout, set_timeout)
         rpc.join_rpc()
 
+    @dist_init
+    def test_rpc_timeouts(self):
+        rpc.set_rpc_timeout(timedelta(milliseconds=1))
+        dst_rank = (self.rank + 1) % self.world_size
+        fut = rpc.rpc_async("worker{}".format(dst_rank), my_sleep_func, args=())
+        with self.assertRaisesRegex(RuntimeError, "Future ran for more than"):
+            fut.wait()
+
+        # future should run to completion if the timeout is longer.
+        dist.barrier()
```

Comment:
Why do we need these barriers? Also, don't we need to initialize the process group if we need these barriers? I don't think `dist_init` initializes the default process group for all backends.

Reply:
I thought there might be issues with different timeouts existing across different workers. Though if this happens, the test will still work, because the timeout on the local worker is always used

Reply:
I see, but we still need to initialize the process group right? This test probably won't work for non process group backends?